internal/controller/pdbwatcher_controller.go

```diff
@@ -102,14 +122,39 @@ func (r *PDBWatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	deployment := &appsv1.Deployment{}
 	err = r.Get(ctx, types.NamespacedName{Name: deploymentName, Namespace: pdbWatcher.Namespace}, deployment)
 	if err != nil {
-		return ctrl.Result{}, err
+		return ctrl.Result{}, err // Error fetching Deployment
 	}
 
+	// Track initial state to detect conflicts
+	initialResourceVersion := deployment.ResourceVersion
```

Comment:
nitpick the prefix initial her is kind of funny. Its the current replicas and resource version because you just refetched it